app/services/post_importer.rb

```diff
@@ -3,23 +3,20 @@ def initialize(url)
     @url = url
   end
 
-  def import(board_id, importer_id, section_id: nil, status: Post.statuses[:complete], threaded: false)
+  def import(params, user:)
     validate_url!
-    validate_duplicate!(board_id) unless threaded
+    validate_duplicate!(params[:board_id]) unless params[:threaded]
     validate_usernames!
 
-    # note that the arg order for this import method does not match the order of ScrapePostJob
-    ScrapePostJob.perform_later(@url, board_id, section_id, status, threaded, importer_id)
+    ScrapePostJob.perform_later(@url, params, user: user)
   end
 
   def self.valid_dreamwidth_url?(url)
     # this is simply checking for a properly formatted Dreamwidth URL
     # errors when actually querying the URL are handled by ScrapePostJob
-    return false if url.blank?
-    return false unless url.include?('dreamwidth')
+    return false if url.blank? || url.exclude?('dreamwidth')
     parsed_url = URI.parse(url)
-    return false unless parsed_url.host
-    parsed_url.host.ends_with?('dreamwidth.org')
+    parsed_url.host&.ends_with?('dreamwidth.org')
```

Comment:
```suggestion
    !!parsed_url.host&.ends_with?('dreamwidth.org')
```
might want this to ensure that `parsed_url.host` being nil returns false rather than nil (to match the `?` on the end of the method name)

Reply:
I'm somewhat confused on how it's ensuring that? how much of this is the !! affecting.

Reply:
it should be the whole thing, `(parsed_url.host&ends_with?('dreamwidth.org')`, which evaluates to `parsed_url.host` if it's nil and otherwise the overall thing (a bool) if it's not, and then applies `!!` to that

```rb
irb(main):001> require 'active_support'
=> true
irb(main):002> require 'active_support/core_ext' # for String#ends_with?
=> true
irb(main):004> x = nil
=> nil
irb(main):006> x&.ends_with?('dreamwidth.org')
=> nil
irb(main):008> !!x&.ends_with?('dreamwidth.org')
=> false
irb(main):010> x = 'url.dreamwidth.org'
=> "url.dreamwidth.org"
irb(main):012> x&.ends_with?('dreamwidth.org')
=> true
irb(main):014> !!x&.ends_with?('dreamwidth.org')
=> true
irb(main):016> x = 'example.com'
=> "example.com"
irb(main):018> x&.ends_with?('dreamwidth.org')
=> false
irb(main):020> !!x&.ends_with?('dreamwidth.org')
=> false
```

Reply:
Huh, I would definitely expect it to just operate on x, but okay.